api/src/transactions.rs

```diff
@@ -959,22 +983,47 @@ impl TransactionsApi {
     async fn get_by_hash(
         &self,
         hash: aptos_crypto::HashValue,
-        ledger_info: &LedgerInfo,
+        storage_ledger_info: &LedgerInfo,
+        internal_ledger_info: Option<&LedgerInfo>,
     ) -> anyhow::Result<Option<TransactionData>> {
         let context = self.context.clone();
-        let version = ledger_info.version();
+        let version = storage_ledger_info.version();
         let from_db =
             tokio::task::spawn_blocking(move || context.get_transaction_by_hash(hash, version))
                 .await
                 .context("Failed to join task to read transaction by hash")?
                 .context("Failed to read transaction by hash from DB")?;
         Ok(match from_db {
-            None => self
-                .context
-                .get_pending_transaction_by_hash(hash)
-                .await?
-                .map(|t| t.into()),
-            _ => from_db.map(|t| t.into()),
+            None => match self.context.get_pending_transaction_by_hash(hash).await? {
```

Comment:
instead of doing db -> mempool -> db, do we want to change it to only do mempool -> db? @msmouse

Reply:
The reason is more likely to be in DB

Reply:
I doubt that, especially when the network is loaded.
I feel we should check mempool first, to make the code simper, and also checking mempool should be much cheaper than DB?

Reply:
I am not sure as well. But this is more likely not to change existing behavior that could introduce some unknown bugs.

Reply:
I'm fine either way (but I lean towards simplicity)

Reply:
anyway, updated to mempool -> db.